shims/spark311/src/main/scala/com/nvidia/spark/rapids/shims/spark311/ParquetCachedBatchSerializer.scala

```diff
@@ -272,8 +272,16 @@ class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
     }
   }
 
-  def isSupportedByCudf(schema: Seq[Attribute]): Boolean = {
-    schema.forall(a => GpuColumnVector.isNonNestedSupportedType(a.dataType))
+  def isSchemaSupportedByCudf(schema: Seq[Attribute]): Boolean = {
+    schema.forall(field => isSupportedByCudf(field.dataType))
+  }
+
+  def isSupportedByCudf(dataType: DataType): Boolean = {
+    dataType match {
+      // TODO: have a case for ArrayType
```

Comment:
nit: I assume this is a follow on issue? If so it might be cleaner to say

```
// TODO: when arrays are supported for cudf writes add it here.
```